pkg/render/intrusion_detection.go

```diff
@@ -1136,14 +1175,255 @@ func (c *intrusionDetectionComponent) intrusionDetectionAnnotations() map[string
+					Name:      ADAPIObjectName,
+					Namespace: IntrusionDetectionNamespace,
+					Labels: map[string]string{
+						"k8s-app": ADAPIObjectName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: ADAPIObjectName,
+					// setting tolerations, nodeselector as the node affinity setting respecting
+					// the control plane tagged node as the centeralized node for anomaly
+					// detection storage
+					Tolerations:      c.cfg.Installation.ControlPlaneTolerations,
+					NodeSelector:     c.cfg.Installation.ControlPlaneNodeSelector,
+					ImagePullSecrets: secret.GetReferenceList(c.cfg.PullSecrets),
+					Volumes: []corev1.Volume{
+						c.cfg.TrustedCertBundle.Volume(),
+						c.cfg.ADAPIServerCertSecret.Volume(),
+						{
+							Name: adAPIStorageVolumeName,
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  ADAPIObjectName,
+							Image: c.adAPIImage,
+							Env:   envVars,
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/health",
+										Port:   intstr.FromInt(adAPIPort),
+										Scheme: corev1.URISchemeHTTPS,
+									},
+								},
+								InitialDelaySeconds: 5,
+								PeriodSeconds:       10,
+								FailureThreshold:    5,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/health",
+										Port:   intstr.FromInt(adAPIPort),
+										Scheme: corev1.URISchemeHTTPS,
+									},
+								},
+								InitialDelaySeconds: 5,
+								PeriodSeconds:       10,
+								FailureThreshold:    5,
+							},
+							Command: []string{"/anomaly-detection-api"},
+							VolumeMounts: []corev1.VolumeMount{
+								c.cfg.TrustedCertBundle.VolumeMount(),
+								c.cfg.ADAPIServerCertSecret.VolumeMount(),
+								{
+									MountPath: adAPIStorageVolumePath,
+									Name:      adAPIStorageVolumeName,
+									ReadOnly:  false,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// AD Detector PodTemplate  with RBAC
+func (c *intrusionDetectionComponent) adDetectorServiceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      adDetectorServiceAccountName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) adDetectorAccessRole() *rbacv1.Role {
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      adDetectorServiceAccountName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{
+					ADJobPodTemplateBaseName,
```

Comment:
I mentioned this just now in the other PR, this should be `detectors.tigera.io` and not what the `ADJobPodTemplateBaseName` refers to.

Reply:
https://github.com/tigera/operator/pull/1931/commits/243589f6f789db15732483f8ba163a1b349e89bc